components/tidb_query_datatype/src/codec/datum.rs

```diff
@@ -1960,7 +1960,7 @@ mod tests {
             ),
             (Datum::Bytes(b"[1, 2, 3]".to_vec()), "[1, 2, 3]"),
             (Datum::Bytes(b"{}".to_vec()), "{}"),
-            (Datum::I64(1), "true"),
```

Comment:
Why it even affects cast...

Reply:
It doesn't, but this test verifies whether the output is expected through `assert_eq :facepalm: .